pkg/client/didexchange/client.go

```diff
@@ -35,6 +35,30 @@ const (
 // ErrConnectionNotFound is returned when connection not found.
 var ErrConnectionNotFound = errors.New("connection not found")
 
+type options struct {
+	routerConnID string
+}
+
+func applyOptions(args ...Opt) *options {
+	opts := &options{}
+
+	for i := range args {
+		args[i](opts)
+	}
+
+	return opts
+}
+
+// Opt represents option function.
+type Opt func(*options)
+
+// WithRouterConnectionID allows you to specify the router connection.
+func WithRouterConnectionID(connID string) Opt {
+	return func(opts *options) {
+		opts.routerConnID = connID
```

Comment:
@soluchok why only one? `opts.routerConnID` could be an array

Reply:
ok, I will add this.

Reply:
I was also thinking of the user injecting a _strategy_ for router selection.

Instead of `WithRouterConnectionID` you would have `WithRouterSelection`. Example:

```go
inv, err := client.CreateInvitation(protocols, oob.WithRouterSelection(
    func(svc mediator.ProtocolService) ([]string, error) {
        // return the router connection IDs
    }
))
```

Or something similar. The output doesn't necessarily have to be connection IDs and the input does not necessarily have to be `mediator.ProtocolService`.

The point is we should allow the user to inject _behaviour_, not values.

Reply:
I don't think it helps. Provide a connection is the easiest way. Also, it works for REST. You can figure out the connection outside. I mean you don’t need a function.

Reply:
> it works for REST

In that case, ok... make it an array of connection IDs. We should definitely not limit the user to just one router.

Reply:
Sure, I'll do that.

Reply:
@llorllale Added possibility to specify multiple connections. If connections are not specified we will add all connections to the `new did doc`. See `getDIDDocAndConnection` function.